projects/ngx-sub-form/src/lib/ngx-sub-form-utils.ts

```diff
@@ -32,6 +35,44 @@ export type FormErrors<FormInterface> = null | Partial<
   }
 >;
 
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>; // can be removed when upgrading ts version
+
+// defaulting to options in form control
+type DefaultSetValueOptions = Parameters<FormControl['setValue']>[1];
+type DefaultPatchValueOptions = Parameters<FormControl['patchValue']>[1];
+
+// tslint:disable-next-line:interface-over-type-literal
+export type TypedControlBase<TValue, TControl extends FormArray | AbstractControl = AbstractControl> = {
+  value: TValue;
+  valueChanges: Observable<TValue>;
+  setValue(
+    value: TValue,
+    options?: TControl extends FormArray ? Parameters<FormArray['setValue']>[1] : DefaultSetValueOptions,
+  ): void;
+  patchValue(
+    value: Partial<TValue>,
+    options?: TControl extends FormArray ? Parameters<FormArray['setValue']>[1] : DefaultPatchValueOptions,
+  ): void;
+};
+
+type UntypedControlBase<TControl extends FormArray | AbstractControl> = Omit<
+  TControl,
+  keyof TypedControlBase<any, TControl>
+>;
+
+export type TypedAbstractControl<TValue> = TypedControlBase<TValue, AbstractControl> &
+  UntypedControlBase<AbstractControl>;
```

Comment:
I know it doesn't change anything there but I'd prefer to have them in a safer order for the override:

```ts
export type TypedAbstractControl<TValue> = UntypedControlBase<AbstractControl> & TypedControlBase<TValue, AbstractControl>;
```